shakenfist/external_api/app.py

```diff
@@ -1130,12 +1130,15 @@ def get(self, node=None):
         if node:
             f.append(partial(images.placement_filter, node))
 
+        retval = []
         for i in images.Images(filters=f):
-            yield i.external_view()
+            retval.append(i.external_view())
+        return retval
 
     @jwt_required
     def post(self, url=None):
         db.add_event('image', url, 'api', 'cache', None, None)
+        images.Image.new(url)
```

Comment:
This line is properly worth a comment. queues.py#133 is probably worth reviewing. Calling `.new()` twice during the image fetch process looks wrong.

We might also need a `queued` state to more accurately describe the image state??

Reply:
Added a comment to explain this. I think initial is sufficient as a state to be honest.